ext/gd/gd.c

```diff
@@ -3545,7 +3545,7 @@ PHP_FUNCTION(imageflip)
 			break;
 
 		default:
-			zend_argument_value_error(2, "must be a valid mode");
+			zend_argument_value_error(2, "must be either GD_FLIP_VERTICAL (1), GD_FLIP_HORINZONTAL (2), or GD_FLIP_BOTH (3)");
```

Comment:
I've realized that explicitly displaying the possible values is much more helpful than the "must be valid" blanket statement.

Reply:
I'd suggest to drop the numeric values `(1)` etc though. These are implementation details.